cluster/saltbase/salt/supervisor/kube-proxy-checker.sh

```diff
@@ -0,0 +1,34 @@
+#!/bin/bash
+
+# Copyright 2015 The Kubernetes Authors All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# This script is intended to start the kube-proxy and then loop until
+# it detects a failure.  It then exits, and supervisord restarts it
+# which in turn restarts the kube-proxy.
+
+/etc/init.d/kube-proxy stop
+/etc/init.d/kube-proxy start
+
+echo "waiting a minute for startup"
+sleep 60
+
+while true; do
+  if ! curl -f -s http://127.0.0.1:10249/healthz > /dev/null; then
```

Comment:
The monit documentation is pretty unclear on which one of the two options it's timeout corresponds to. I originally read it as the connect timeout, but I agree that max time is closer to what we actually want (rather than just trying to exactly clone the monit configs).